### PR TITLE
Add doc section on debug compiler flags

### LINKED ISSUE #301: add a debug argument to load_all()
I've spent quite some time trying to understand why, when compiling some C++ code with `devtools::load_all()`, I would always get the `g -O0` flags (among others) at the very end, thus overriding my preferred settings in `~/.R/Makevars` or `src/Makevars`. Eventually I've worked out that these are appended by `pkgbuild::compile_dll()`, which is called by `pkgload::load_all()`.

I think the current default makes sense, but it would be nice if `pkgload::load_all()` had a `debug` flag that gets passed to `pkgbuild::compile_dll()`, so that the debug options could be disabled easily. Alternatively, `load_all()` could accept a `...` argument and pass that to `compile_dll()` (this seems to me an inferior solution though).

### PR DESCRIPTION
**Pull request by mcol**
This fixes #301 as suggested in #302. See if the wording is clear enough, suggestions welcome!

### CODE REVIEW

File: R/load.R [hunk 1]
```diff
@@ -64,6 +64,16 @@
 #'    behavior when loading an installed package with [library()], and can
 #'    be useful for checking for missing exports.
 #'
+#' # Controlling the debug compiler flags
+#'
+#' `load_all()` delegates to [pkgbuild::compile_dll()] to perform the actual
+#' compilation, during which by default some debug compiler flags are
+#' appended. If this is undesired (for example, if you would like to produce
+#' an optimized build), you can opt out by setting the `pkg.build_extra_flags`
```
Comment (lionel-) [inline R/load.R:72]:
```suggestion
#' appended. If you would like to produce an optimized build instead),
#' you can opt out by setting the `pkg.build_extra_flags`
```

**Review by lionel- — COMMENTED** [review]
Looking good, just a small suggestion

File: R/load.R (hunk 1 above)
Comment (mcol) [reply R/load.R]: Thanks for the suggestion! I didn't apply it directly for a mismatched bracket. :)

**Comment by lionel-** [comment]
Thank you for this!

**Pull request merged**